plugins/woocommerce/src/Internal/ProductFilters/FilterData.php

```diff
@@ -0,0 +1,349 @@
+<?php
+/**
+ * Class for filter counts.
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\ProductFilters;
+
+use WC_Cache_Helper;
+use Automattic\WooCommerce\Internal\ProductFilters\Interfaces\QueryClausesGenerator;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for filter counts.
+ */
+class FilterData {
+	/**
+	 * Instance of QueryClauses.
+	 *
+	 * @var QueryClausesGenerator
+	 */
+	private $query_clauses;
+
+	/**
+	 * Initialize dependencies.
+	 *
+	 * @internal
+	 *
+	 * @param QueryClauses $query_clauses Instance of QueryClauses.
+	 *
+	 * @return void
+	 */
+	final public function init( QueryClauses $query_clauses ): void {
+		$this->set_query_clauses( $query_clauses );
+	}
+
+	/**
+	 * Allow setting the clauses provider at run time.
+	 *
+	 * @param QueryClausesGenerator $query_clauses Instance of QueryClauses.
+	 *
+	 * @return void
+	 */
+	final public function set_query_clauses( QueryClausesGenerator $query_clauses ): void {
+		$this->query_clauses = $query_clauses;
+	}
+
+	/**
+	 * Get price data for current products.
+	 *
+	 * @param array $query_vars The WP_Query arguments.
+	 * @return object
+	 */
+	public function get_filtered_price( array $query_vars ) {
+		$pre_filter_counts = $this->pre_get_filter_counts( 'price', $query_vars, );
+
+		if ( isset( $pre_filter_counts ) ) {
+			return $pre_filter_counts;
+		}
+
+		$transient_key = $this->get_transient_key( $query_vars, 'price' );
+		$cached_data   = $this->get_cache( $transient_key );
+
+		if ( ! empty( $cached_data ) ) {
+			return $cached_data;
+		}
+
+		global $wpdb;
+
+		add_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10, 2 );
+		add_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$query_vars['no_found_rows']  = true;
+		$query_vars['posts_per_page'] = -1;
+		$query_vars['fields']         = 'ids';
+		$query                        = new \WP_Query();
+
+		$query->query( $query_vars );
+		$product_query_sql = $query->request;
+
+		remove_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10 );
+		remove_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$price_filter_sql = "
+		SELECT min( min_price ) as min_price, MAX( max_price ) as max_price
+		FROM {$wpdb->wc_product_meta_lookup}
+		WHERE product_id IN ( {$product_query_sql} )
+		";
```

Comment:
Should it be using `$wpdb->prepare`? 

```suggestion
		$price_filter_sql = $wpdb->prepare(
                "
                    SELECT MIN(min_price) as min_price, MAX(max_price) as max_price
                    FROM {$wpdb->wc_product_meta_lookup}
                    WHERE product_id IN ( %s )
                ",
                $product_query_sql
);
```

Reply:
We can't. `$product_query_sql` is already prepared. Running through another preparation causes the query to return nothing.
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/2ec92e83-2f8d-48aa-a6e0-cf05ce1231c3">

Reply:
@thealexandrelara We can't. `$product_query_sql` is already prepared. Running through another prepare causes the query to return nothing. Same for your other comments, too.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/2ec92e83-2f8d-48aa-a6e0-cf05ce1231c3">